backend/src/data/repositories/user/user.repository.ts

```diff
@@ -31,6 +31,24 @@ class User {
     return user ?? null;
   }
 
+  async getUserPermissions(id: number): Promise<PermissionM[] | null> {
+    const permissions = await this.#UserModel
+      .query()
+      .select(
+        'groups:permissions.id',
+        'groups:permissions.name',
+        'groups:permissions.key',
+      )
+      .joinRelated('groups.permissions')
+      .where('users.id', id);
+
+    if (permissions.length === 0) {
+      return null;
+    }
+
+    return permissions as unknown as PermissionM[];
```

Comment:
use `castTo` instead

Reply:
I'm not quite sure what that means

Reply:
https://vincit.github.io/objection.js/api/types/#castto - it may perform type casts inside the generated query, but also you can provide type for query result by generic